packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2413,11 +2419,7 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
   /// Paste text from [Clipboard].
   @override
   Future<void> pasteText(SelectionChangedCause cause) async {
-    if (widget.readOnly) {
-      return;
-    }
-    final TextSelection selection = textEditingValue.selection;
-    if (!selection.isValid) {
+    if (!_allowPaste) {
```

Comment:
Should we immediately return here? If that's the case then the assert in `_pasteText` will likely never be hit. I wonder if we should fail loudly (assert), and return silently, so in debug mode we will get the failure but in release mode it fails silently. Before this change we always just failed silently. This comment also applies to the other `_allowPaste` usage.

```dart
  if (processedText != null && _allowPaste) {
    _pasteText(SelectionChangedCause.toolbar, processedText);
  } 
```

Reply:
My goal with this assert was to express the fact that `_pasteText` should not be called without having checked `_allowPaste` before.
In this PR, `_pasteText` is called twice and we correctly checked `_allowPaste` before so the assert will never be hit. My concern was that if someone add a call to `_pasteText`, they should be aware of the requirement to call `_allowPaste` before.

On second thoughts, this assert is not great at all because one can never read this line nor hit the assert while testing and so forget to call `_allowPaste`.

I did that to keep the exact same logic as before (the check is done before querying the clipboard). I think a more robust way is to replace the assert with a call to `_allowPaste` and return silently (as you suggested). And if we want to return also before querying the clipboard (to keep the same logic as before this PR) it means that `_allowPaste` will be call twice (one in `pasteText` before querying the clipboard and one in `_pasteText` to avoid having someone calling this method without checking `_allowPaste`).

@Renzo-Olivares The question for you is do you prefer to keep the current logic (and doing so having two calls to `_allowPaste`) or to slightly change the logic by querying the clipboard before checking `_allowPaste`.
Not sure I was clear here, I will file an update tomorrow, it will be easier to discuss.

Reply:
I have updated the PR. I removed the assert I added previously and I added a call to `_allowPaste` in `_pasteText`.